src/fides/api/service/connectors/mongodb_connector.py

```diff
@@ -71,7 +71,7 @@ def test_connection(self) -> Optional[ConnectionTestStatus]:
             )
             if default_auth_db:
                 db = client[default_auth_db]
-                db.collection_names()
+                db.list_collection_names()
```

Comment:
As per the [pymongo 4.0 migration docs](https://pymongo.readthedocs.io/en/stable/migrate-to-pymongo4.html#database-collection-names-is-removed), `collection_names` is removed and we need to use `list_collection_names` instead.